### PR TITLE
Remove duplicate parse IDs code, avoid coreinternal dependency

### PR DESCRIPTION
**Pull request by bogdandrutu**
The main reason to do this, is to allow coreinternal/processor/filter* to use ottl.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: pkg/ottl/contexts/internal/ids/ids.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+package ids // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ids"
+
+import (
+	"encoding/hex"
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func ParseSpanID(spanIDStr string) (pcommon.SpanID, error) {
```
Comment (TylerHelmuth) [inline pkg/ottl/contexts/internal/ids/ids.go:24]: Do these need to be moved to their own pacakge?  Could they live in `ottlcommon` in an `ids.go`?

Comment (bogdandrutu) [reply pkg/ottl/contexts/internal/ids/ids.go:24]: It can. Moved.

**Review by TylerHelmuth — APPROVED** [review]

File: pkg/ottl/contexts/internal/ottlcommon/ids_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+package ottlcommon
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseSpanIDError(t *testing.T) {
```
Comment (evan-bradley) [inline pkg/ottl/contexts/internal/ottlcommon/ids_test.go:23]: Could we have some tests to validate the happy path?

Comment (bogdandrutu) [reply pkg/ottl/contexts/internal/ottlcommon/ids_test.go:23]: Happy to review a PR :))

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**